### DISCUSSION TITLE
Can I call the built-in commands provided by vscode in the terminal?

### DISCUSSION THREAD

**Discussion by dddssw** [microsoft/vscode-discussions#2436] [Extension Development QnA] [answered]
I have an idea. generate template code through cli and insert the code into the currently edited file. Is this feasible? The difficulty I encountered was how to get the address of the currently edited file.

**Comment by alefragnani** [comment, marked as answer]
Hi @dddssw ,

There is no `argument` to the `code` CLI that neither exposes the currently edited file, nor to edit the text of the currently editing file. 

So, in order to accomplish your idea, I think you will need to create not only a new CLI, but also an extension, and make them talk to each other. Similar to [GistPad CLI](https://github.com/lostintangent/gistpad-cli).

Hope this helps

**Reply by dddssw** [reply]
Thank you, I think it's a bit troublesome to create a new plugin. Can I use onDidChangeActiveTextEditor to remember and update the currently edited file? I'm not sure if this will have a significant impact on performance

**Reply by alefragnani** [reply]
I think you can use `onDidChangeActiveTextEditor`, for sure. About the performance impact, shouldn't be a problem. The event and the editing process tends to be unnoticeable. 

Just be aware of the processing requirements of your extension, and validate it using large files, if those are valid scenarios.

**Reply by dddssw** [reply]
I see, thank
